### PR TITLE
Login and register forms

### PR DESCRIPTION
**Pull request by Miszo97**
**header updated**

Now it has login and sign up buttons
New components were added:

1. for login: https://mui.com/material-ui/getting-started/templates/sign-in/
2. for registration : https://mui.com/material-ui/getting-started/templates/sign-up/

### CODE REVIEW

File: frontend/src/components/Header.js [hunk 1]
```diff
@@ -1,38 +1,53 @@
...
+          >
+            <Link href="/signin">
+              <Button sx={{ my: 2, color: "white", display: "block" }}>
+                {" "}
+                Login
+              </Button>
+            </Link>
+            <Link href="/signup">
+              <Button sx={{ my: 2, color: "white", display: "block" }}>
+                {" "}
```
[the remark this answers is not in the document]
Comment (Miszo97) [reply frontend/src/components/Header.js]: fixed

File: frontend/src/components/Header.js [hunk 2]
```diff
@@ -1,38 +1,53 @@
...
+          <Box
+            sx={{
+              flexGrow: 0,
+              backgroundColor: "inherit",
+              display: { xs: "none", md: "flex" },
+            }}
+          >
+            <Link href="/signin">
+              <Button sx={{ my: 2, color: "white", display: "block" }}>
+                {" "}
```
[the remark this answers is not in the document]
Comment (Miszo97) [reply frontend/src/components/Header.js]: fixed

File: frontend/src/components/SignUp.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  );
+}
+
+const theme = createTheme();
+
+export default function SignUp() {
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    console.log({
```
Comment (tomkeee) [inline frontend/src/components/SignUp.tsx:40]: and here

File: frontend/src/components/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  );
+}
+
+const theme = createTheme();
+
+export default function SignIn() {
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    console.log({
```
Comment (tomkeee) [inline frontend/src/components/SignIn.js:40]: do we have to console.log it here?

File: frontend/src/components/SignIn.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  );
+}
+
+const theme = createTheme();
+
+export default function SignIn() {
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    console.log({
```
Comment (tomkeee) [inline frontend/src/components/SignIn.tsx:40]: same here

File: frontend/src/components/SignUp.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  );
+}
+
+const theme = createTheme();
+
+export default function SignUp() {
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    console.log({
```
Comment (tomkeee) [inline frontend/src/components/SignUp.js:40]: and here

File: frontend/src/components/SignUp.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+                  id="password"
+                  autoComplete="new-password"
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <FormControlLabel
+                  control={
+                    <Checkbox value="allowExtraEmails" color="primary" />
+                  }
+                  label="I want to receive inspiration, marketing promotions and updates via email."
```
Comment (tomkeee) [inline frontend/src/components/SignUp.js:118]: Is it necessary  to have this kind of label ?

File: frontend/src/components/SignUp.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+          <Box
+            component="form"
+            noValidate
+            onSubmit={handleSubmit}
+            sx={{ mt: 3 }}
+          >
+            <Grid container spacing={2}>
+              <Grid item xs={12} sm={6}>
+                <TextField
+                  autoComplete="given-name"
```
Comment (tomkeee) [inline frontend/src/components/SignUp.js:73]: Our User model does not have a first name and last name field. How about having just username field instead?

File: frontend/src/components/SignUp.tsx [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+function Copyright(props: any) {
+  return (
+    <Typography
+      variant="body2"
+      color="text.secondary"
+      align="center"
+      {...props}
+    >
+      {"Copyright © "}
+      <Link color="inherit" href="https://mui.com/">
```
Comment (tomkeee) [inline frontend/src/components/SignUp.tsx:25]: and here with the copyright

File: frontend/src/App.js [hunk 10]
```diff
@@ -13,14 +15,14 @@ function App() {
       <main>
         <Container>
           <Routes>
-          < Route path="/news" element = {<NewsScreen />}/>
-          < Route path="/settings" element = {<SearchSettings />}/>
-
+            <Route path="/news" element={<NewsScreen />} />
+            <Route path="/settings" element={<SearchSettings />} />
+            <Route path="/signin" element={<SignIn />} />
```
Comment (tomkeee) [inline frontend/src/App.js:20]: I think that we shouldn't include Header component in login page (redirections to user's Search Settings and News)

File: frontend/src/components/SignIn.js [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+  return (
+    <Typography
+      variant="body2"
+      color="text.secondary"
+      align="center"
+      {...props}
+    >
+      {"Copyright © "}
+      <Link color="inherit" href="https://mui.com/">
+        Your Website
```
Comment (tomkeee) [inline frontend/src/components/SignIn.js:26]: We should also update this copyright, it suggests that our website name is Your Website - it should be Favourite News.
Also It redirects to https://mui.com/

File: frontend/src/components/SignUp.js [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+  return (
+    <Typography
+      variant="body2"
+      color="text.secondary"
+      align="center"
+      {...props}
+    >
+      {"Copyright © "}
+      <Link color="inherit" href="https://mui.com/">
+        Your Website
```
Comment (tomkeee) [inline frontend/src/components/SignUp.js:26]: same here with copyright

File: frontend/src/components/SignUp.tsx [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+                  id="password"
+                  autoComplete="new-password"
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <FormControlLabel
+                  control={
+                    <Checkbox value="allowExtraEmails" color="primary" />
+                  }
+                  label="I want to receive inspiration, marketing promotions and updates via email."
```
Comment (tomkeee) [inline frontend/src/components/SignUp.tsx:118]: and here with the label

**Review by tomkeee — CHANGES_REQUESTED** [review]

**Review by tomkeee — APPROVED** [review]
Great job!

**Pull request merged**